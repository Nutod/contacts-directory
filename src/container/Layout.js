import React, { Component } from "react";
import { Switch, Route } from "react-router-dom";
import SignUp from "../components/auth/SignUp";
import SignIn from "../components/auth/SignIn";
import AddNewContact from "../components/contact/AddNewContact";
import Account from "../components/profile/Account";
import Profile from "../components/profile/Profile";
import ContactList from "../components/contact/ContactList";

// TODO: Default should load  Account if authenticated else, redirect to auth page
// TODO: Change Icons
// TODO: Add Responsiveness
export default class Layout extends Component {
	render() {
		return (
			<Switch>
				<Route path="/auth/sign-up" component={SignUp} />
				<Route path="/auth/sign-in" component={SignIn} />
				<Route path="/add-new" component={AddNewContact} />
				<Route path="/account" component={Account} />
				<Route path="/profile" component={Profile} />
				<Route path="/contact-list" component={ContactList} />
				<Route path="/" exact component={SignIn} />
			</Switch>
		);
	}
}
