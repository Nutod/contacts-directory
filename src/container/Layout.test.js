import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import Layout from "./Layout";

Enzyme.configure({ adapter: new Adapter() });

describe("<Layout />", () => {
	const wrapper = shallow(<Layout />);
	it("Should return default true", () => {
		expect(true).toBe(true);
		console.log(wrapper.debug());
	});

	it("Should match snapshot", () => {
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
