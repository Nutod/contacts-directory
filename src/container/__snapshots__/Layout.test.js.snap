// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> Should match snapshot 1`] = `
<Switch>
  <Route
    component={[Function]}
    path="/auth/sign-up"
  />
  <Route
    component={[Function]}
    path="/auth/sign-in"
  />
  <Route
    component={[Function]}
    path="/add-new"
  />
  <Route
    component={[Function]}
    path="/account"
  />
  <Route
    component={[Function]}
    path="/profile"
  />
  <Route
    component={[Function]}
    path="/contact-list"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
</Switch>
`;
